src/utils/bintools.ts

```diff
@@ -284,11 +285,40 @@ export default class BinTools {
   addressToString = (hrp:string, chainid:string, bytes:Buffer)
   :string => `${chainid}-${bech32.encode(hrp, bech32.toWords(bytes))}`;
 
-  stringToAddress = (address:string):Buffer => {
-    const parts:Array<string> = address.trim().split('-');
-    if(parts[1].startsWith("0x") || parts[1].match(/^[0-9A-F]+$/i)){
-      return Buffer.from(parts[1].replace("0x", ""), "hex");
+  stringToAddress = (address: string, hrp?: string): Buffer => {
+    if(address.substring(0, 2) === "0x") {
+      // ETH-style address
+      if(utils.isAddress(address)) {
+        return Buffer.from(address.replace("0x", ""), "hex");
```

Comment:
Probably makes sense to replace `address.replace("0x", "")` with `address.substring(2,)`
Otherwise any string that starts `0x` can have an arbitrary number of `0x` in it.
e.g. 0x0x0x

Reply:
Oh, does `replace` only do the first occurrence? nvm then

Reply:
Personally I still think `address.substring(2,)` is a little cleaner but if it works as-is then fine by me

Reply:
```ts
const p = '0x0xyo';
console.log(p.replace('0x', ''));
"0xyo"
```

We should be good w/ what we have.

Reply:
`replace` replaced w/ `substring`